src/process/tracking.rs

```diff
@@ -0,0 +1,296 @@
+                    info.lock().unwrap().idx += 1;
+
+                    // Store a reference to the TrackingInfo in the ovs_queue tracking table.
+                    self.ovs_queue_tracking
+                        .insert(enqueue.queue_id, info.clone());
+
+                    Self::insert_info(event, info)?;
+                }
+                RecvUpcall(recv) => {
+                    let info = self.lookup_ovs_queue(recv.queue_id)?;
+                    info.lock().unwrap().idx += 1;
+                    Self::insert_info(event, &info)?;
+                }
+                Operation(op) => {
+                    let info = self.lookup_ovs_queue(op.queue_id)?;
+                    info.lock().unwrap().idx += 1;
+                    Self::insert_info(event, &info)?;
+                }
+                Action(act) => match act.queue_id {
+                    Some(queue_id) => {
+                        // This action event came from an upcall. Restore the tracking id of the
+                        // original packet.
+                        let info = self.lookup_ovs_queue(queue_id)?;
+                        info.lock().unwrap().idx += 1;
+
+                        // Add an entry in the skb tracking table so that futre non-ovs events also
+                        // get the tracking id from the original (upcalled) packet.
+                        if let Some(skb) =
+                            event.get_section::<SkbTrackingEvent>(ModuleId::SkbTracking)
+                        {
+                            self.skb_tracking.insert(skb.tracking_id(), info.clone());
+                        }
+
+                        Self::insert_info(event, &info)?;
+                    }
+                    None => {
+                        self.process_skb(event)?;
+                    }
+                },
+                Undefined => bail!("Cannot track undefined ovs event"),
+            }
+        } else {
+            // It's not an OVS event, try skb-only tracking.
+            self.process_skb(event)?;
+        }
+        Ok(())
+    }
+
+    // Insert TrackingInformation to an event.
+    fn insert_info(event: &mut Event, info: &Arc<Mutex<TrackingInfo>>) -> Result<()> {
+        let info = info.lock().unwrap().clone();
+        if let Some(info_section) = event.get_section::<TrackingInfo>(ModuleId::Tracking) {
+            if *info_section != info {
+                bail!("Event already has info section {info_section:?} and does not match computed {info:?}")
+            }
+        } else {
+            event.insert_section(ModuleId::Tracking, Box::new(info))?;
+        }
+
+        Ok(())
+    }
+
+    // Add tracking information to an event based on skb tracking id.
+    fn process_skb(&mut self, event: &mut Event) -> Result<Arc<Mutex<TrackingInfo>>> {
+        let skb = event
+            .get_section::<SkbTrackingEvent>(ModuleId::SkbTracking)
+            .ok_or_else(|| anyhow!("No skb tracking id found"))?;
```

Comment:
This error ends up being caught by `sort`, logged at  debug level and the event is printed in it's own series (i.e: an untracked event)

Reply:
Well I don't mind silencing it at the source.

Reply:
I'm thinking that we should still throw an error if we have inconsistent tracking, no?
I mean, if we have an OVS upcall with queue_id but without skb-tracking, it can only mean skb-tracking module was not activated. In that case we should just fail as adding information would just mess things up even more, don't you  think?

Reply:
How about the latest update?
https://github.com/retis-org/retis/compare/4cec697cd5a5706e34bc6fe070a3b2edd013ed96..c39dc24b77f69a8f9a90a82178729b39ba7cd85e